src/dataflow-types/src/types.rs

```diff
@@ -667,6 +667,7 @@ pub struct TailSinkConnector {
     pub tx: comm::mpsc::Sender<Vec<Row>>,
     pub frontier: Antichain<Timestamp>,
     pub strict: bool,
+    pub emit_timestamps: bool,
```

Comment:
if you choose to rename to `PROGRESS` this variable would also be great to rename to `emit_progress`